### DISCUSSION TITLE
cleanReport: alert

### DISCUSSION THREAD

**Discussion by ghost** [mrbungle64/ecovacs-deebot.js#292] [Q&A] [answered]
Is there a way to find out what the reason is when there is an alert? And / or is there a way to trap/capture the other messages like reminders and stuff.

**Comment by mrbungle64** [comment]
@cactusss 

I don't really understand the question. Please be more specific with your question 😉

**Comment by ghost** [comment]
Well, if my Deebot gets trapped, I get a message on my phone (from Ecovacs Mobile app) that it is trapped, and I need to set it free. Or that it is 'off the floor'.

I would like my app to be able to send the same message; family members don't have the Ecovacs app, but I am able to send them the messages through my own app. All I can do now (AFAIK) is detect an alert (cleanReport: alert) but I can't figure out how to detect what caused the alert (deebot is trapped, or off the floor).

Also, I get reminders to change the mopping cloth, or clean the dustbin. It would be awesome if I could also handle those messages with my app instead of the Ecovacs app.

**Reply by mrbungle64** [reply]
@cactusss 

There's an "error" event, but the library doesn't handle the specific "alert" of the "CleanInfo" (CleanReport) message yet.
In [this](https://github.com/mrbungle64/ecovacs-deebot.js/blob/master/library/errorCodes.json) file you can find the already known error codes.

**Reply by mrbungle64** [reply]
@At-Home-with-Automation 

> I cant seem to locat the "error" value within the Clean Report, is there another call that covers the error codes?

You can use the Error event that I already mentioned:
https://github.com/mrbungle64/ecovacs-deebot.js/wiki/Events#errors

But maybe I don't understand what you mean exactly 😉

**Reply by mrbungle64** [reply]
@At-Home-with-Automation 

I still don't understand why you expect to receive an error code when sending a CleanReport call ...

That is what I posted on Feb 01:

> There's an "error" event, but the library doesn't handle the specific "alert" of the "CleanInfo" (CleanReport) message yet.

**Comment by ghost** [comment]
Ah yeah. I handled the vacbot.on('error') incorrctly.

I now have this:

`this.vacbot.on('Error', (error) => {`
`	if (error.search('NoError') == -1) {`
`               // do some stuff`
`	}`
`});`

The returned error is only the text message ("HostHang: Robot is off the floor" for instance) which is okay. But if at all possible it would be nicer to get the errorcode instead of (or besides) the error message. But I can work with this also.

Thanks again Sascha!

**Reply by mrbungle64** [reply]
@cactusss 

> But if at all possible it would be nicer to get the errorcode instead of (or besides) the error message.

For the error code you can use the "ErrorCode" event.

**Comment by ghost** [comment, marked as answer]
Thanks! That'll work.

[a comment by an automated account is not shown]
